app/models/user.rb

```diff
@@ -140,14 +140,22 @@ def currently_in_exam?
     exams.any? { |e| e.in_progress_for? self }
   end
 
+  def profile_picture
+    (self.avatar || self)&.image_url
+  end
+
   private
 
   def set_uid!
     self.uid ||= email
   end
 
   def init
-    self.image_url ||= "user_shape.png"
+    self.avatar = random_avatar unless profile_picture.present?
+  end
+
+  def random_avatar
+    Avatar.order('RANDOM()').first
```

Comment:
This doesn't cause performance issues with few (< thousands) rows. I don't think we'll have more than a couple dozens, so it should be okay.